### PR TITLE
Add support for Office 365 and mixed Outlook versions

### PR DESCRIPTION
**Pull request by eshenfield**
This PR covers most Office 365 emails and also some cases of emails between various Outlook versions.

I've also re-organized the Microsoft splitter-finding code to define a variety of queries in one place (so we can add others easily) and then programmatically check for the existence of any of those queries, taking the first one in the DOM as the top-most splitter and removing everything after it.

Other things included in this PR:

- [x] Add test for new example HTML file

- [x] Update version of jsdom for tests

- [x] This addition also does not handle emails between providers very well (for example: send email from Office 365 to Gmail, reply from Gmail, reply from Office 365 -- gmail quote is stripped but Office 365 quoted text remains. This may not be done in this PR but I'd like to at least add a TODO or a note of this

### CODE REVIEW

**Comment by eshenfield** [comment]
All tests pass!

screen shot 2017-11-22 at 1 33 34 pm

File: src/htmlPlaner.coffee [hunk 1]
```diff
@@ -198,26 +198,55 @@ exports.replaceBreakTagsWithLineFeeds = (emailDocument) ->
...
+      possibleSplitterElements.push splitterElement
 
-  if splitterResult.length > 0
-    splitterElement = splitterResult[0]
-    # Outlook 2010
-    if splitterElement.parentElement? && splitterElement == splitterElement.parentElement.children[0]
-      splitterElement = splitterElement.parentElement
+  return null unless possibleSplitterElements.length
+  # Find the earliest splitter in the DOM to remove everything after it
+  earliestThing = possibleSplitterElements.sort(compareByDomPosition)[0]
```
Comment (l8on) [inline src/htmlPlaner.coffee]: This `earliestThing` assignment is not used. This line can likely be removed.

**Review by l8on — COMMENTED** [review]
I like the organization of all the string constants used by Microsoft generated emails.
There is a tiny bit of code cleanup keeping this from approval, but once that's done this will LGTM.

File: src/htmlPlaner.coffee (hunk 1 above)
Comment (eshenfield) [reply src/htmlPlaner.coffee]: 😂  whoops, yup -- definitely leftover from debugging. Thanks for catching!

**Review by l8on — APPROVED** [review]

**Pull request merged**